src/session/user.rs

```diff
@@ -146,12 +147,64 @@ mod imp {
 
             let avatar = obj.avatar();
             let user_expression = gtk::ConstantExpression::new(obj);
-            let full_name_expression = super::User::full_name_expression(&user_expression);
-            full_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
+            let display_name_expression = super::User::display_name_expression(&user_expression);
+            display_name_expression.bind(avatar, "display-name", gtk::NONE_WIDGET);
         }
     }
 }
 
+fn last_seen_time_ago(unix: i32) -> String {
+    let datetime_now = glib::DateTime::new_now_local().unwrap();
+    let last_seen = glib::DateTime::from_unix_utc(unix.into())
+        .unwrap()
+        .to_local()
+        .unwrap();
+
+    let seconds = datetime_now.difference(&last_seen) / 1_000_000;
+    let minutes = seconds / 60;
+    let hours = minutes / 60;
+    let days = hours / 24;
+
+    if minutes == 0 {
+        gettext("last seen just now")
+    } else if minutes == 1 {
+        gettext("last seen a minute ago")
+    } else if minutes < 60 {
+        gettext!("last seen {} minutes ago", minutes)
+    } else if minutes < 120 {
+        gettext("last seen an hour ago")
+    } else if (datetime_now.day_of_month() == last_seen.day_of_month()) & (hours < 24) {
+        gettext!("last seen {} hours ago", hours)
+    } else if (datetime_now.day_of_month() == last_seen.add_days(1).unwrap().day_of_month())
+        & (days < 2)
+    {
+        last_seen
+            .format(&gettext("last seen yesterday at %l:%M %p"))
+            .unwrap()
+            .to_string()
+    } else if days < 7 {
+        last_seen
+            .format(&gettext("last seen on %A at %l:%M %p"))
+            .unwrap()
+            .to_string()
+    } else if days < 14 {
+        last_seen
+            .format(&gettext("last seen last %A at %l:%M %p"))
+            .unwrap()
+            .to_string()
+    } else if datetime_now.year() == last_seen.year() {
+        last_seen
+            .format(&gettext("last seen on %d %b at %l:%M %p"))
+            .unwrap()
+            .to_string()
+    } else {
+        last_seen
+            .format(&gettext("last seen %d.%m.%Y at %l:%M %p"))
+            .unwrap()
+            .to_string()
+    }
+}
```

Comment:
I put this at the end. You could reduce the lines of codes a lot if you exploit rust's expression syntax. This removes a lot of duplicate code lines.
```rust
    if minutes == 0 {
        gettext("last seen just now")
    } else if minutes == 1 {
        gettext("last seen a minute ago")
    } else if minutes < 60 {
        gettext!("last seen {} minutes ago", minutes)
    } else if minutes < 120 {
        gettext("last seen an hour ago")
    } else if (datetime_now.day_of_month() == last_seen.day_of_month()) & (hours < 24) {
        gettext!("last seen {} hours ago", hours)
    } else {
        last_seen
            .format(&if (datetime_now.day_of_month()
                == last_seen.add_days(1).unwrap().day_of_month())
            {
                gettext("last seen yesterday at %l:%M %p")
            } else if days < 7 {
                gettext("last seen on %A at %l:%M %p")
            } else if days < 14 {
                gettext("last seen last %A at %l:%M %p")
            } else if datetime_now.year() == last_seen.year() {
                gettext("last seen on %d %b at %l:%M %p")
            } else {
                gettext("last seen %d.%m.%Y at %l:%M %p")
            })
            .unwrap()
            .to_string()
    }
```

Reply:
I've got busy week, I'll commit your suggestion and hopefully you'll be able to merge it